lib/services/semaphore.go

```diff
@@ -0,0 +1,604 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/utils"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// SemaphoreKindSessionControl is the semaphore kind used by
+// the Concurrent Session Control feature.
+const SemaphoreKindSessionControl = "sessctl"
```

Comment:
Do we have any other kind of semaphore?

Reply:
Also, probably need a better name for this @benarent.

Reply:
After today we are going to remove this for now, and review when we come back to locking / terminating sessions.